DESIGN.md

```diff
@@ -0,0 +1,56 @@
+# TLS Provider Design
+
+The TLS Provider offers a small surface area compared to other providers (like
+[AWS](https://registry.terraform.io/providers/hashicorp/aws/latest),
+[Google](https://registry.terraform.io/providers/hashicorp/google/latest),
+[Azure](https://registry.terraform.io/providers/hashicorp/azurerm/latest), ...),
+and focuses on covering the needs of working with entities like
+keys and certificates, that are part of
+[Transport Security Layer](https://en.wikipedia.org/wiki/Transport_Layer_Security).
+
+Below we have a collection of _Goals_, _Patterns_ and _Taboos_: they represent the guiding principles applied during
+the development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+ 
+## Goals
+
+* Support [cryptography](https://en.wikipedia.org/wiki/Cryptography) _primitives_ necessary to Terraform configurations
+* Provide managed resourced and data sources to manipulate and interact with **Keys, Certificates and Certificate Requests**
+* Support file and data formats widely used in the industry: this means sticking with open and well documented standards
```

Comment:
We should confirm this is an actual product goal before committing this here as it feels overly broad. 👍  For example, this provider has historically only implemented certain, specific IETF RFCs that fit well into the use of PEM encoding given it was attempted to be made a de-facto Terraform convention. It may be desired that certain file and data formats are considered out of scope for this provider and instead would need to be implemented in other providers.

Reply:
I'm going to try to remove the vagueness and instead list what we actually support.

Additionally, I want to add a sub-section to `Goals` where we say:

```markdown
### About formats and key algorithms

Cryptography and security are an evolving and changing subject; for this reason the set of supported technologies for
this provider will need to be reassessed over time by the maintenance team, while also evaluating incoming
feature requests.
```